mp/src/game/server/client.cpp

```diff
@@ -847,52 +848,65 @@ CON_COMMAND( say_team, "Display player message to team" )
 {
-	CBasePlayer *pPlayer = ToBasePlayer( UTIL_GetCommandClient() ); 
-	if ( pPlayer 
-		&& (gpGlobals->maxClients == 1 || sv_cheats->GetBool()) 
-		&& args.ArgC() >= 2
-		&& !pPlayer->IsObserver() )
+	const auto pCmdName = "give_weapon";
+	char *substring = nullptr;
+	if (Q_strstr(pPartial, pCmdName) && strlen(pPartial) > strlen(pCmdName) + 1)
 	{
-		char item_to_give[ 256 ];
-		Q_strncpy( item_to_give, args[1], sizeof( item_to_give ) );
-		Q_strlower( item_to_give );
+		substring = (char *)pPartial + strlen(pCmdName) + 1;
+	}
 
-		// Don't allow regular users to create point_servercommand entities for the same reason as blocking ent_fire
-		if ( !Q_stricmp( item_to_give, "point_servercommand" ) )
+	int current = 0;
+	// Skip over weapon_none
+	for (int weaponID = WEAPON_PISTOL; weaponID < WEAPON_MAX; weaponID++)
+	{
+		if (!g_pGameModeSystem->GetGameMode()->WeaponIsAllowed((WeaponID_t)weaponID))
+			continue;
+
+	    if ((!substring || !substring[0]) || Q_stristr(g_szWeaponNames[weaponID], substring))
+	    {
+			char command[COMMAND_COMPLETION_ITEM_LENGTH];
+			Q_snprintf(command, COMMAND_COMPLETION_ITEM_LENGTH, "%s %s", pCmdName, g_szWeaponNames[weaponID]);
+	        Q_strncpy(commands[current], command, COMMAND_COMPLETION_ITEM_LENGTH);
+			current++;
+	    }
+	}
+
+	return current;
+}
+
+CON_COMMAND_F_COMPLETION(give_weapon, "Gives the player a weapon.", 0, WeaponCompletion)
+{
+	const auto pPlayer = ToCMOMPlayer(UTIL_GetCommandClient());
+	if (pPlayer && !pPlayer->IsObserver() && args.ArgC() == 2)
+	{
+		WeaponID_t foundID = WEAPON_NONE;
+		for (int weaponID = WEAPON_PISTOL; weaponID < WEAPON_MAX; weaponID++)
 		{
-			if ( engine->IsDedicatedServer() )
-			{
-				// We allow people with disabled autokick to do it, because they already have rcon.
-				if ( pPlayer->IsAutoKickDisabled() == false )
-					return;
-			}
-			else if ( gpGlobals->maxClients > 1 )
-			{
-				// On listen servers with more than 1 player, only allow the host to create point_servercommand.
-				CBasePlayer *pHostPlayer = UTIL_GetListenServerHost();
-				if ( pPlayer != pHostPlayer )
-					return;
-			}
+		    if (!Q_strnicmp(g_szWeaponNames[weaponID], args.Arg(1), 64))
+		    {
+	            foundID = (WeaponID_t)weaponID;
+				break;
+		    }
 		}
 
-		// Dirty hack to avoid suit playing it's pickup sound
-		if ( !Q_stricmp( item_to_give, "item_suit" ) )
+		if (foundID == WEAPON_NONE)
 		{
-			pPlayer->EquipSuit( false );
+		    Warning("Could not give weapon with name %s, weapon not found!\n", args.Arg(1));
 			return;
 		}
 
-		string_t iszItem = AllocPooledString( item_to_give );	// Make a copy of the classname
-		pPlayer->GiveNamedItem( STRING(iszItem) );
+	    if (!g_pGameModeSystem->GetGameMode()->WeaponIsAllowed(foundID))
+	    {
+	        Warning("The weapon %s is not allowed in this gamemode!\n", args.Arg(1));
+			return;
+	    }
+
+		pPlayer->GiveWeapon(foundID);
```

Comment:
Tab/space mixing

Reply:
Resolved in https://github.com/momentum-mod/game/pull/624/commits/79a39fad8fd283aa27e934a67aad6c8806ef7947